website/static/js/nodesDelete.js

```diff
@@ -127,68 +117,48 @@ var NodesDeleteViewModel = function(node) {
 
     self.page = ko.observable(self.SELECT);
 
-    self.pageTitle = ko.computed(function() {
-        if (self.nodeType === 'project'){
-            return {
-                select: 'Delete Project',
-                confirm: 'Delete Project and Components'
-            }[self.page()];
-        }
+    self.pageTitle = ko.computed(function () {
+        return {
+            select: 'Delete ' + self.nodeType,
+            confirm: 'Delete ' + self.nodeType === 'project' ? 'project and ' : '' + 'components'
+        }[self.page()];
+    });
 
-        if (self.nodeType === 'component'){
-            return {
-                select: 'Delete Component',
-                confirm: 'Delete Components'
-            }[self.page()];
-        }
+    self.message = ko.computed(function () {
+        var message = 'It looks like your ' + self.nodeType + ' has components within it. To delete this ' +
+          self.nodeType + ', you must also delete all child components.';
 
-    });
+        var confirm_message = 'The following ' + (self.nodeType === 'project' ? 'project and ' : '') + 'components will be deleted.';
 
-    self.message = ko.computed(function() {
         if (self.page() === self.CONFIRM) {
-            if (self.nodeType === 'project')
-                return 'The following project and components will be deleted';
-
-            if (self.nodeType === 'component')
-                return 'The following components will be deleted';
+            return confirm_message;
         }
 
-        if (self.nodeType === 'project')
-            return {
-                select: 'It looks like your project has components within it. To delete this project, you must also delete all child components',
-                confirm: 'The following project and components will be deleted.'
-            }[self.page()];
-
-        if (self.nodeType === 'component')
-            return {
-                select: 'It looks like your componet has components within it. To delete this component, you must also delete all child components',
-                confirm: 'The following components will be deleted.'
-            }[self.page()];
+        return {
+            select: message + (self.isPreprint ? self.preprintMessage : ''),
```

Comment:
This preprintMessage is only thrown if the parent node is a preprint, but what if you are bulk deleting nodes and a child node is a preprint?  I don't know a quick way to recursively check all the children and make sure there are no preprints, but I think the user needs a warning that they may be inadvertently deleting a preprint.